### PR TITLE
Map methods in AbstractBlock and subclasses

### PR DESCRIPTION
**Pull request by LemmaEOF**
These names might need some work, I did my best to figure out how to be descriptive with them.

- `prepare` is called inside of `PistonBlock#move` and `world#setBlockState`. It has no implementation in `AbstractBlock` and no block overrides its behavior.
- `getSidesShape` is used in `Block#topCoversMediumSquare`, `Block#sideCoversSmallSquare`, and `Block#isSideSolidFullSquare`. No block overrides its behavior.
- `getVisualShape` is used in `RayTraceContext.ShapeType.VISUAL`. No block overrides its behavior.
- `AbstractBlock#blockVisionPredicate`, `Settings#blockVisionPredicate`, `Settings#blockVision`, and `AbstractBlockState#shouldBlockVision` are all used client-side for whether a user's screen should be blocked when suffocating. I'm not sure how it's different from `inWallOverlay`.

### CODE REVIEW

File: mappings/net/minecraft/block/AbstractBlock.mapping [hunk 1]
```diff
@@ -9,19 +9,27 @@ CLASS net/minecraft/class_4970 net/minecraft/block/AbstractBlock
...
+	FIELD field_23164 velocityMultiplier F
 	FIELD field_23165 slowDownMultiplier F
 	METHOD <init> (Lnet/minecraft/class_4970$class_2251;)V
 		ARG 1 settings
 	METHOD method_16841 getOffsetType ()Lnet/minecraft/class_4970$class_2250;
 	METHOD method_17454 createScreenHandlerFactory (Lnet/minecraft/class_2680;Lnet/minecraft/class_1937;Lnet/minecraft/class_2338;)Lnet/minecraft/class_3908;
 	METHOD method_19286 onProjectileHit (Lnet/minecraft/class_1937;Lnet/minecraft/class_2680;Lnet/minecraft/class_3965;Lnet/minecraft/class_1676;)V
 	METHOD method_22358 canBucketPlace (Lnet/minecraft/class_2680;Lnet/minecraft/class_3611;)Z
+	METHOD method_25959 getSidesShape (Lnet/minecraft/class_2680;Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;)Lnet/minecraft/class_265;
+	METHOD method_26159 getVisualShape (Lnet/minecraft/class_2680;Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;Lnet/minecraft/class_3726;)Lnet/minecraft/class_265;
```
Comment (liach) [inline mappings/net/minecraft/block/AbstractBlock.mapping:33]: Hmm, mind restore the parameters?

File: mappings/net/minecraft/block/AbstractBlock.mapping [hunk 2]
```diff
@@ -9,19 +9,27 @@ CLASS net/minecraft/class_4970 net/minecraft/block/AbstractBlock
...
+	METHOD method_26161 hasBlockEntity ()Z
+	METHOD method_26162 getDropTableID ()Lnet/minecraft/class_2960;
+	METHOD method_26163 getMaterialColor ()Ljava/util/function/Function;
 	METHOD method_8389 asItem ()Lnet/minecraft/class_1792;
 	METHOD method_9498 hasComparatorOutput (Lnet/minecraft/class_2680;)Z
 	METHOD method_9505 getOpacity (Lnet/minecraft/class_2680;Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;)I
 	METHOD method_9506 emitsRedstonePower (Lnet/minecraft/class_2680;)Z
 	METHOD method_9514 scheduledTick (Lnet/minecraft/class_2680;Lnet/minecraft/class_3218;Lnet/minecraft/class_2338;Ljava/util/Random;)V
 	METHOD method_9516 canPathfindThrough (Lnet/minecraft/class_2680;Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;Lnet/minecraft/class_10;)Z
+	METHOD method_9517 prepare (Lnet/minecraft/class_2680;Lnet/minecraft/class_1936;Lnet/minecraft/class_2338;I)V
```
Comment (liach) [inline mappings/net/minecraft/block/AbstractBlock.mapping:61]: what does this do...

Comment (LemmaEOF) [reply mappings/net/minecraft/block/AbstractBlock.mapping:61]: Absolutely nothing! It’s called in `world#setBlockState` and `PistonBlock#move`, has no default impl, and no overrides.

File: mappings/net/minecraft/block/AbstractBlock.mapping (hunk 1 above)
Comment (LemmaEOF) [reply mappings/net/minecraft/block/AbstractBlock.mapping:33]: Oh right, params.

Comment (LemmaEOF) [reply mappings/net/minecraft/block/AbstractBlock.mapping:33]: Actually, what parameters do you want me to restore?

Comment (liach) [reply mappings/net/minecraft/block/AbstractBlock.mapping:33]: Hmm, somehow all the parameter names are lost... Like `world`, `pos`, etc

File: mappings/net/minecraft/block/AbstractBlock.mapping (hunk 2 above)
Comment (liach) [reply mappings/net/minecraft/block/AbstractBlock.mapping:61]: Interesting, looks good then.

**Review by liach — APPROVED** [review]
@Boundarybreaker Mind fix the merge conflict real quick?

File: mappings/net/minecraft/block/AbstractBlock.mapping (hunk 1 above)
Comment (LemmaEOF) [reply mappings/net/minecraft/block/AbstractBlock.mapping:33]: Ah yeah, I can fix that real quick.

Comment (LemmaEOF) [reply mappings/net/minecraft/block/AbstractBlock.mapping:33]: all the parameters are in!

File: mappings/net/minecraft/block/AbstractBlock.mapping [hunk 3]
```diff
@@ -83,6 +217,9 @@ CLASS net/minecraft/class_4970 net/minecraft/block/AbstractBlock
 		FIELD field_23184 blockVisionPredicate Lnet/minecraft/class_4970$class_4973;
 		FIELD field_23185 postProcessPredicate Lnet/minecraft/class_4970$class_4973;
 		FIELD field_23186 inWallOverlayPredicate Lnet/minecraft/class_4970$class_4973;
+		METHOD <init> (Lnet/minecraft/class_3614;Ljava/util/function/Function;)V
+			ARG 1 material
+			ARG 2 materialColor
```
Comment (liach) [inline mappings/net/minecraft/block/AbstractBlock.mapping]: Hmm, is this a `materialColorFactory`

File: mappings/net/minecraft/block/AbstractBlock.mapping [hunk 4]
```diff
@@ -15,51 +15,185 @@ CLASS net/minecraft/class_4970 net/minecraft/block/AbstractBlock
...
+		ARG 1 state
+		ARG 2 world
+		ARG 3 pos
+		ARG 4 player
 	METHOD method_9612 neighborUpdate (Lnet/minecraft/class_2680;Lnet/minecraft/class_1937;Lnet/minecraft/class_2338;Lnet/minecraft/class_2248;Lnet/minecraft/class_2338;Z)V
+		ARG 1 state
+		ARG 2 world
+		ARG 3 pos
+		ARG 4 block
+		ARG 5 posFrom
```
Comment (liach) [inline mappings/net/minecraft/block/AbstractBlock.mapping]:
```suggestion
		ARG 5 fromPos
```

File: mappings/net/minecraft/block/AbstractBlock.mapping [hunk 5]
```diff
@@ -15,51 +15,185 @@ CLASS net/minecraft/class_4970 net/minecraft/block/AbstractBlock
...
+		ARG 6 notify
 	METHOD method_9615 onBlockAdded (Lnet/minecraft/class_2680;Lnet/minecraft/class_1937;Lnet/minecraft/class_2338;Lnet/minecraft/class_2680;Z)V
+		ARG 1 state
+		ARG 2 world
+		ARG 3 pos
+		ARG 4 stateFrom
```
Comment (liach) [inline mappings/net/minecraft/block/AbstractBlock.mapping]:
```suggestion
		ARG 4 oldState
```

File: mappings/net/minecraft/block/AbstractBlock.mapping [hunk 6]
```diff
@@ -139,76 +311,217 @@ CLASS net/minecraft/class_4970 net/minecraft/block/AbstractBlock
...
 		METHOD method_26189 getDroppedStacks (Lnet/minecraft/class_47$class_48;)Ljava/util/List;
+			ARG 1 builder
 		METHOD method_26190 getRenderingSeed (Lnet/minecraft/class_2338;)J
+			ARG 1 pos
 		METHOD method_26191 getStateForNeighborUpdate (Lnet/minecraft/class_2350;Lnet/minecraft/class_2680;Lnet/minecraft/class_1936;Lnet/minecraft/class_2338;Lnet/minecraft/class_2338;)Lnet/minecraft/class_2680;
+			ARG 1 direction
+			ARG 2 state
+			ARG 3 world
+			ARG 4 pos
+			ARG 5 posFrom
```
Comment (liach) [inline mappings/net/minecraft/block/AbstractBlock.mapping]:
```suggestion
			ARG 5 fromPos
```

File: mappings/net/minecraft/block/AbstractBlock.mapping [hunk 7]
```diff
@@ -222,4 +535,13 @@ CLASS net/minecraft/class_4970 net/minecraft/block/AbstractBlock
 			METHOD <init> (Lnet/minecraft/class_2680;)V
 				ARG 1 state
 	CLASS class_4972 TypedContextPredicate
+		METHOD test test (Lnet/minecraft/class_2680;Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;Ljava/lang/Object;)Z
+			ARG 1 state
+			ARG 2 world
+			ARG 3 pos
+			ARG 4 type
 	CLASS class_4973 ContextPredicate
+		METHOD test test (Lnet/minecraft/class_2680;Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;)Z
```
Comment (liach) [inline mappings/net/minecraft/block/AbstractBlock.mapping]:
```suggestion
		METHOD test (Lnet/minecraft/class_2680;Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;)Z
```

File: mappings/net/minecraft/block/AbstractBlock.mapping [hunk 8]
```diff
@@ -222,4 +535,13 @@ CLASS net/minecraft/class_4970 net/minecraft/block/AbstractBlock
 			METHOD <init> (Lnet/minecraft/class_2680;)V
 				ARG 1 state
 	CLASS class_4972 TypedContextPredicate
+		METHOD test test (Lnet/minecraft/class_2680;Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;Ljava/lang/Object;)Z
```
Comment (liach) [inline mappings/net/minecraft/block/AbstractBlock.mapping]:
```suggestion
		METHOD test (Lnet/minecraft/class_2680;Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;Ljava/lang/Object;)Z
```

**Review by liach — COMMENTED** [review]
Looks fairly good!
1. we use suffix naming, so "from position" I think (it is a "pos" not a "from")
2. need to unmap the `test` methods, dunno why enigma mapped them

File: mappings/net/minecraft/block/AbstractBlock.mapping (hunk 3 above)
Comment (LemmaEOF) [reply mappings/net/minecraft/block/AbstractBlock.mapping]: That's a good point. I think we should name the field/arg `materialColorFactory` or `colorFactory` and have the method still be named `materialColor`

File: mappings/net/minecraft/block/AbstractBlock.mapping [hunk 9]
```diff
@@ -9,49 +9,191 @@ CLASS net/minecraft/class_4970 net/minecraft/block/AbstractBlock
...
 	METHOD method_9571 getCullingShape (Lnet/minecraft/class_2680;Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;)Lnet/minecraft/class_265;
+		ARG 1 state
+		ARG 2 world
+		ARG 3 pos
 	METHOD method_9572 getComparatorOutput (Lnet/minecraft/class_2680;Lnet/minecraft/class_1937;Lnet/minecraft/class_2338;)I
+		ARG 1 state
+		ARG 2 world
+		ARG 3 pos
 	METHOD method_9575 getAmbientOcclusionLightLevel (Lnet/minecraft/class_2680;Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;)F
+		ARG 1 tate
```
Comment (YanisBft) [inline mappings/net/minecraft/block/AbstractBlock.mapping]: `state`

**Review by YanisBft — APPROVED** [review]

**Review by Juuxel — APPROVED** [review]

**Review by liach — APPROVED** [review]

**Pull request merged**